lua/telescope/utils.lua

```diff
@@ -284,4 +284,244 @@ function utils.strcharpart(str, nchar, charlen)
+        -- edge_wincol returns the left or right border wincol instead depending on 'maximum' param
+            -- get edge pos for wincol start (most left) and end (most right) by line from original positions
+            -- flipping is required to know what direction to resolve to
+            wincol_start = utils.edge_wincol({l_flip and l_end or l_start,          -- get l_end if flipped else original l_start
+                                              c_flip and c_end - 1 or c_start - 1},  -- get c_end if flipped else original c_start
+                                              wincol_start,                            -- resolve for wincol_start
+                                              c_flip)                                -- flipped (true) or not (false), resolve to true=right, left=false
+            wincol_end = utils.edge_wincol({l_flip and l_start or l_end,            -- resolve conversely in opposite direction as above
+                                            c_flip and c_start - 1 or c_end - 1},
+                                            wincol_end,
+                                            not c_flip)
+            -- properly set left and right edge wincol
+            wincol_left = math.min(wincol_start, wincol_end)
+            wincol_right = math.max(wincol_start, wincol_end)
+        return l_start, c_start, l_end, c_end, wincol_left, wincol_right
+    else
+      -- char- or linewise selection
+      if l_start > l_end or (l_start == l_end and c_end < c_start) then
+        return l_end, c_end, l_start, c_start
+      end
+      return l_start, c_start, l_end, c_end
+    end
+end
+
+-- Retrieve wincol from pos
+-- wincol (see :help wincol) denotes the visual column of the current cursor position
+-- Since wincol can only be retrieved for the current position,
+-- the cursor has to be set prior
+-- @param pos table position to retrieve wincol for
+-- @param start_pos table optional position to jump back to
+function utils.get_wincol(pos, start_pos)
+    start_pos = start_pos or false
+    -- pos {line, col} - {1, 0}-indexed
+    vim.api.nvim_win_set_cursor(0, pos)
+
+    local wincol = vim.fn.wincol()
+    if start_pos then
+        vim.api.nvim_win_set_cursor(0, start_pos)
+    end
+    return wincol
+end
+
+-- Get maximum, possibly synthetic wincol for position
+-- Consider a multi-width char: 全, wincol solely returns the first visual column
+-- edge_wincol returns the left or right border wincol instead depending on parameters
+-- @param pos table {row, column} with {1, 0} indexed values
+-- @param wincol integer initial wincol prior to finding (synthetic) maximum
+-- @param maximum boolean true resolves towards right, false resolves towards left
+function utils.edge_wincol(pos, wincol, maximum)
+    maximum = maximum or false
+    local max_col = utils.get_wincol({pos[1], 2^31-1})
+    if wincol >= max_col then
+        return max_col
+    end
+    if wincol == 1 then
+        return 1
+    end
```

Comment:
not sure if this breaks anything but it maybe better to have one parent if expression, parentheses are not required but make it more readable and accept additional `and/or`
```lua
if wincol then 
  return (wincol >= max_col) and max_col or 1 
end 
```